wfsim/core.py

```diff
@@ -387,10 +387,20 @@ def er(size,config, singlet_triplet_delays, phase):
         timings *= 1 / (1 - np.random.uniform(0, 1, size)) - 1
         timings = np.clip(timings, 0, config['maximum_recombination_time'])
         size_primary = len(timings[primary])
-        timings[primary] += singlet_triplet_delays(
-            size_primary, config['s1_ER_primary_singlet_fraction'],config,phase)
-        timings[~primary] += singlet_triplet_delays(
-            size - size_primary, config['s1_ER_secondary_singlet_fraction'],config,phase)
+
+        timings[primary] += Pulse.singlet_triplet_delays(
+            size_primary, config['s1_ER_primary_singlet_fraction'], config, phase)
+
+        # Correct for the recombination time
+        # For the non-exponential distribution: see Kubota 1979, solve eqn 2 for n/n0.
+        # Alternatively, see Nest V098 source code G4S1Light.cc line 948
+        timings[~primary] *= 1 / (-1 + 1 / np.random.uniform(0, 1, size - size_primary))
+        # Update max recombine time in the nT fax config
+        config['maximum_recombination_time'] = 1000
```

Comment:
perhaps do `config.setdefault('maximum_recombination_time', 1000)`, Is this the right place by the way?

Reply:
Unfortunately, it has been set in all our config files, but it seems that the original value is a bit too tight. We can remove this line but remember to change in the other files.